includes/admin/class-amp-post-meta-box.php

```diff
@@ -422,4 +429,79 @@ public function preview_post_link( $link ) {
 
 		return $link;
 	}
+
+	/**
+	 * Add a REST API field to display the AMP status on AMP supported post types.
+	 *
+	 * @return void
+	 */
+	public function add_rest_api_fields() {
+		register_rest_field(
+			AMP_Post_Type_Support::get_post_types_for_rest_api(),
+			self::REST_ATTRIBUTE_NAME,
+			[
+				'get_callback'    => [ $this, 'amp_status_get_callback' ],
+				'update_callback' => [ $this, 'amp_status_update_callback' ],
+				'schema'          => [
+					'description' => __( 'AMP status', 'amp' ),
+					'type'        => 'boolean',
+				],
+			]
+		);
+	}
+
+	/**
+	 * Display the AMP status of a post via the REST API.
+	 *
+	 * @param array $post_data Post data.
+	 * @return bool Whether AMP is enabled on post.
+	 */
+	public function amp_status_get_callback( $post_data ) {
+		$status = $this->sanitize_status( get_post_meta( $post_data['id'], self::STATUS_POST_META_KEY, true ) );
+
+		if ( '' === $status ) {
+			$post              = get_post( $post_data['id'] );
+			$status_and_errors = self::get_status_and_errors( $post );
+
+			if ( isset( $status_and_errors['status'] ) ) {
+				$status = $status_and_errors['status'];
+			}
+		}
+
+		return self::ENABLED_STATUS === $status;
+	}
+
+	/**
+	 * Update the AMP status of a post via the REST API.
+	 *
+	 * @param bool    $is_enabled Whether AMP is enabled.
+	 * @param WP_Post $post       Post being updated.
+	 * @return void|WP_Error Returns an instance of `WP_Error` if the post failed to be updated.
+	 */
+	public function amp_status_update_callback( $is_enabled, $post ) {
```

Comment:
Should this have a check to see if `$post->post_type` is among `AMP_Post_Type_Support::get_post_types_for_rest_api()`?

Reply:
I'll be limiting the `amp_enabled` field to only  `posts`, so yes this should be checking to see if `$post->post_type` is `post`.

Reply:
Addressed in https://github.com/ampproject/amp-wp/pull/4689/commits/49992b009c6da80bc13fc40d6410231d3e3eedbb.